clipmenu

```diff
@@ -57,11 +57,25 @@ else
 fi
 
 [[ $chosen_line ]] || exit 1
-file=$cache_dir/$(cksum <<< "$chosen_line")
-[[ -f "$file" ]] || exit 2
+
+files=()
+while IFS= read -r line; do
+    file=$cache_dir/$(cksum <<< "$line")
+    files+=("$file")
+done <<< "$chosen_line"
+
+paste=()
+for file in "${files[@]}"; do
+    contents=$(cat "$file")
+    paste+=("$contents")
+done
+
+paste=$( IFS=$'\n'; echo "${paste[*]}" )
+
+ [[ $paste ]] || exit 2
 
 for selection in clipboard primary; do
-    xsel --logfile /dev/null -i --"$selection" < "$file"
+    echo "$paste" | awk 'NR>1{print PREV} {PREV=$0} END{printf("%s",$0)}' | xsel --logfile /dev/null -i --"$selection"
```

Comment:
Wait, why are you explicitly mangling trailing newlines? That's literally something we try as hard as possible to preserve, so doing that is a non-starter.

Reply:
This was specifically to remove the trailing newline that `echo` creates. I'm not sure if `cat` would behave the same but I'll test it.